server/lib/s3backup/__init__.py

```diff
@@ -189,6 +183,35 @@ def head_bucket(self, Bucket=None):
     def list_objects(self, **kwargs):
         return self.connector.list_objects(**kwargs)
 
+    # response decoding - tbh is the header returned
+    # by get_tarball_header().
+    @staticmethod
+    def s3_md5(tbh):
+        if 'Metadata' in tbh and tbh['Metadata']:
+            if 'md5' in tbh['Metadata']:
+                return tbh['Metadata']['md5']
+            else:
+                return None
+        else:
+            return tbh['ETag'].strip('"')
```

Comment:
Is an `ETag` value the same as a MD5 value?  This seems confusing.

Reply:
Not always: it depends on whether it's a large object or not.

Reply:
I moved the comment but expanded it also: let me know what you think of it.